mmcls/models/backbones/convvit.py

```diff
@@ -0,0 +1,381 @@
+                    drop_rate=drop_rate,
+                    drop_path_rate=dpr[global_layer_idx],
+                    act_cfg=act_cfg)
+                if stage_idx >= 2: # transformer stage
+                    layer_class = TransformerEncoderLayer
+                    _layer_cfg.update(
+                        num_heads=num_heads,
+                        qkv_bias=qkv_bias,
+                        norm_cfg=tx_norm_cfg,
+                    )
+                    self.pos_embed = nn.Parameter(
+                        torch.zeros(1, *stage_img_size, embed_dims[stage_idx]))
+                else: # conv stage
+                    layer_class = ConvBlock
+                    layer_conv_norm_cfg = conv_norm_cfg.copy()
+                    _layer_cfg.update(
+                        norm_cfg=layer_conv_norm_cfg,
+                        )
+                _layer_cfg.update(layer_cfgs[global_layer_idx])
+                blocks.append(layer_class(**_layer_cfg))
+            self.blocks.append(nn.ModuleList(blocks))
+        
+        self._register_load_state_dict_pre_hook(self._prepare_pos_embed)
+
+        self.final_norm = final_norm
+        if final_norm:
+            self.norm1_name, norm1 = build_norm_layer(
+                tx_norm_cfg, embed_dims[-1], postfix=1)
+            self.add_module(self.norm1_name, norm1)
+
+        self.interpolate_mode = interpolate_mode
+        self.out_indices = (
+                [out_indices] if isinstance(out_indices, int) else out_indices)
+        self.out_indices = [
+                x if x >= 0 else sum(num_layers) + x for x in self.out_indices]
+        self.drop_after_pos = (
+            nn.Dropout(p=drop_rate) if drop_rate > 0. else nn.Identity())
+
+    @property
+    def norm1(self):
+        return getattr(self, self.norm1_name)
+
+    def init_weights(self):
+        super().init_weights()
+        if not (isinstance(self.init_cfg, dict) and \
+                self.init_cfg['type'] == 'Pretrained'):
+            trunc_normal_(self.pos_embed, std=0.02)
+
+    def _prepare_pos_embed(self, state_dict, prefix, *args, **kwargs):
+        num_stages = len(self.blocks)
+        name = prefix + 'pos_embed'
+        if name not in state_dict:
+            return
+
+        ckpt_pos_embed_shape = state_dict[name].shape
+        if self.pos_embed.shape != ckpt_pos_embed_shape:
+            from mmcv.utils import print_log
+            logger = get_root_logger()
+            print_log(
+                f'Resize pos_embed.{i} shape from {ckpt_pos_embed_shape} '
+                f'to {self.pos_embeds[i].shape}.',
+                logger=logger)
+            state_dict[name] = resize_pos_embed(state_dict[name],
+                ckpt_pos_embed_shape[1:-1],
+                self.pos_embeds[i].shape[1:-1],
+                self.interpolate_mode, 0)
+
+    def forward(self, x):
+        outs = []
+
+        for stage_idx in range(len(self.blocks)):
+            stage_type = 'c' if stage_idx < 2 else 't'
```

Comment:
Maybe ``c`` denotes ``convolution`` and ``t``denotes ``transformer``, I guesss. Better choose another more explicit name.